sklearn/preprocessing/tests/test_encoders.py

```diff
@@ -553,6 +553,58 @@ def test_ordinal_encoder_raise_missing(X):
         ohe.transform(X)
 
 
+def test_ordinal_encoder_handle_unknowns_string():
+    enc = OrdinalEncoder(handle_unknown='use_encoded_value', unknown_value=-2)
+    X_fit = np.array([['a', 'x'], ['b', 'y'], ['c', 'z']], dtype=object)
+    X_trans = np.array([['c', 'xy'], ['bla', 'y'], ['a', 'x']], dtype=object)
+    enc.fit(X_fit)
+
+    X_trans_enc = enc.transform(X_trans)
+    exp = np.array([[2, -2], [-2, 1], [0, 0]], dtype='int64')
+    assert_array_equal(X_trans_enc, exp)
+
+    X_trans_inv = enc.inverse_transform(X_trans_enc)
+    inv_exp = np.array([['c', None], [None, 'y'], ['a', 'x']], dtype=object)
+    assert_array_equal(X_trans_inv, inv_exp)
+
+
+def test_ordinal_encoder_handle_unknowns_numeric():
+    enc = OrdinalEncoder(handle_unknown='use_encoded_value',
+                         unknown_value=-999)
+    X_fit = np.array([[1, 7], [2, 8], [3, 9]], dtype=object)
+    X_trans = np.array([[3, 12], [23, 8], [1, 7]], dtype=object)
```

Comment:
```suggestion
@pytest.mark.parametrize('dtype', [float, int])
def test_ordinal_encoder_handle_unknowns_numeric(dtype):
    enc = OrdinalEncoder(handle_unknown='use_encoded_value',
                         unknown_value=-999)
    X_fit = np.array([[1, 7], [2, 8], [3, 9]], dtype=dtype)
    X_trans = np.array([[3, 12], [23, 8], [1, 7]], dtype=dtype)
```

(I know it is a little strange to encoding floats and ints, but these dtypes are support in `OrdinalEnoder`)

Reply:
Nice catch. In fact, I forgot an astype in the inverse transform.